src/Symfony/Bundle/FrameworkBundle/Debug/TraceableEventDispatcher.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Debug;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\Console\Debug\CliRequest;
+use Symfony\Component\Console\Event\ConsoleCommandEvent;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpKernel\Debug\TraceableEventDispatcher as BaseTraceableEventDispatcher;
+use Symfony\Component\Stopwatch\Stopwatch;
+
+/**
+ * @internal
+ *
+ * @author Jules Pietri <jules@heahprod.com>
+ */
+final class TraceableEventDispatcher extends BaseTraceableEventDispatcher
+{
+    public function __construct(
+        EventDispatcherInterface $dispatcher,
+        Stopwatch $stopwatch,
+        LoggerInterface $logger = null,
+        private readonly ?RequestStack $requestStack = null
+    ) {
+        parent::__construct($dispatcher, $stopwatch, $logger, $requestStack);
+    }
+
+    /**
+     * Starts a stopwatch section before a command runs to profile listeners.
+     */
+    protected function beforeDispatch(string $eventName, object $event): void
+    {
+        if (!class_exists(ConsoleEvents::class) || !$event instanceof ConsoleCommandEvent) {
+            parent::beforeDispatch($eventName, $event);
+
+            return;
+        }
+
+        $request = $this->requestStack->getCurrentRequest();
+
+        if (!$request instanceof CliRequest || $request->command !== $event->getCommand()) {
+            return;
+        }
+
+        $request->attributes->set('_stopwatch_token', substr(hash('sha256', uniqid(mt_rand(), true)), 0, 6));
+        $this->stopwatch->openSection();
```

Comment:
is it legit that we don't have a corresponding afterDispatch that's call stopSection? the parent class has one

Reply:
Yes, in HTTP context we stop the section after dispatching `kernel.response` and save the profile during `kernel.terminate`.
There are not enough events on CLI context, so I close section and save profile during `console.terminate`.